File: Year 2/Algorithms/Data Structures/0.3/Main.cs
using System;
using System.IO;

class Program
{
    static void Main()
    {
        using (StreamReader fin = new StreamReader("input.txt"))
        using (StreamWriter fout = new StreamWriter("output.txt"))
        {
            long n = long.Parse(fin.ReadLine());
            long i = 0;

            while (n != 0)
            {
                if (n % 2 == 1)
                {
                    fout.WriteLine(i);
                }
                i++;
                n /= 2;
            }
        }
    }
}

File: Year 2/Algorithms/Data Structures/0.3/main.py
with open("input.txt", "r") as fin, open("output.txt", "w") as fout:
    n = int(fin.readline())
    i = 0
    while n:
        if n % 2:
            fout.write(f"{i}\n")
        i += 1
        n //= 2